app/services/account.service.ts

```diff
@@ -33,3 +49,93 @@ export const getAllAccounts = async (
 
   return formattedResult;
 };
+
+export const verifyPaystack = async (reference: string) => {
+  try {
+    const response = await axios.get(`https://api.paystack.co/transaction/verify/${reference}`, {
+      headers: {
+        Authorization: `Bearer ${env.paystack_secret}`,
+      },
+    })
+    return response.data;
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+export const updateDepositTransaction = async (
+  accountId: string,
+  userId: string
+) => {
+  /* 
+  Verify Reference: The system verifies the transaction reference with Paystack to ensure its validity and retrieves the transaction amount.
+  */
+  const existingTxn = await getTransactionByUserId(userId);
+ 
+  const reference = existingTxn.reference;
+
+  const paystackResponse: any = await verifyPaystack(reference);
+
+  if (!paystackResponse.status || paystackResponse.data.status !== 'success') {
+    throw new BadUserRequestError(
+      'Deposit transaction failed. Please try again'
+    );
+  }
+
+  /**
+   * Check Transaction Status: The system checks the transaction status in the database to ensure it is not already finalised.
+   */
+
+  if (existingTxn.status !== 'pending') {
+    throw new BadUserRequestError(
+      'This transaction has already been confirmed'
+    );
+  }
+  /**
+   * Process Transaction: If the status is valid, the system updates the transaction status and deposits the amount into the user's account using database increments.
+   */
+
+  const account = await getAccount(accountId);
+
+  const currentBalance = parseFloat(account.balance);
+  const depositAmount = parseFloat(existingTxn.amount);
+  const newBalance = currentBalance + depositAmount;
```

Comment:
You can't do this when working with money, because when you store value in a variable before you update the DB some things might have happened outside your own values. Do this instead

https://knexjs.org/guide/query-builder.html#increment

Reply:
Its called ACID compliance in databases you can read up on it